services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt

```diff
@@ -0,0 +1,49 @@
+package fi.hel.haitaton.hello.db
+
+import mu.KotlinLogging
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.sql.Connection
+import java.sql.DriverManager
+import java.util.ArrayList
+
+@Component
+@RestController
+@RequestMapping("/api")
+
+class DbController {
+    private val logger = KotlinLogging.logger { }
+
+    @Autowired
+    lateinit var dbConfigProperties: DbConfigProperties
+
+    @GetMapping("/tablenames")
+    fun getAllTables(): String {
+
+        val dbUrl: String = dbConfigProperties.appDatasourceUrl + "?" +
+            "user=" + dbConfigProperties.appDatasourceUsername +
+            "&password=" + dbConfigProperties.appDatasourcePassword
+
+        val con = DriverManager.getConnection(dbUrl)
```

Comment:
Is there a reason to use DriverManager and not DataSource?  From DriverManager docs:
_NOTE: The DataSource interface, new in the JDBC 2.0 API, provides another way to connect to a data source. The use of a DataSource object is the preferred means of connecting to a data source._
Afaik, that "new" is by now ancient history (DataSource came in java 1.4)
(Well, this is just an example API, so anything goes, I guess.)

Reply:
Yes, datasource might be better. The problem there was that if I deternmine spring.datasource to the application.properties, it connects automatically and running fails with no db. So, I didn't want the starting of the app to fail if no db is yet available (as is the situation in our cloud service at the moment). But yeah, definately we should re-check this when we know better..